CMakeLists.txt

```diff
@@ -63,8 +63,16 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
-
-find_package(Z3 4.8.0)
+set(LATEST_Z3_VERSION "4.8.12")
+if (TESTS AND NOT EMSCRIPTEN)
```

Comment:
Why `NOT EMSCRIPTEN`? Does this break the emscripten build?

Reply:
I just add this because we add `tests` folder with that condition; if we don't build `test` directory  we don't need the latest version of z3.

Reply:
ok. It's good then. Perhaps `EMSCRIPTEN=ON`+ `TEST=ON` should just be an error but if it isn't then I guess this condition makes sense.

I'd just add a comment above the condition explaining this: "NOTE: With EMSCRIPTEN=ON we do not build code from test/ dir and that code is our only reason to require latest Z3."

Reply:
OK, I'll add the comment to make it clear.